run.sh

```diff
@@ -50,23 +54,47 @@ if [ -f /share/tesla_ble_mqtt/private.pem ]; then
  delete_legacies
  exit 1
 fi
-if [ ${TESLA_VIN-} ]; then
- log_error "Using depecated configuration parameters --> Exiting."
- log_error "Fix config and restart. If you see this message again, please raise an issue"
+if [ ! -z ${TESLA_VIN} ]; then
+ log_fatal "Using deprecated configuration parameters --> Exiting."
+ log_fatal "Fix config and restart. If you see this message again, please raise an issue"
  exit 1
 fi
 
+
+ble_addr_count=0
+while ble_mac in $BLE_MAC_LIST; do
+  ble_addr_count=$(expr $ble_addr_count + 1)
+  BLE_MAC${ble_addr_count}=$ble_mac
+  PRESENCE_EXPIRE_TIME${ble_addr_count}=9999999999
+  log_debug "Adding $ble_mac to the list, count $ble_addr_count"
+done
+
+vin_count=0
+while vin in $TESLA_VIN_LIST; do
+  vin_count=$(expr $vin_count + 1)
+  BLE_LN${vin_count}=$(tesla_vin2ble_ln $vin)
+  TESLA_VIN${vin_count}=$vin
+  log_debug "Adding $vin to the list, count $vin_count"
+done
+
+if [ $vin_count -eq $mac_addr_count ]; then
+  log_debug "Fantastic, we have $vin_count VIN(s) and $ble_addr_count BLE MAC Addr"
+else
+  log_error "VIN count $vin_count differs from  BLE MAC Addr count $ble_addr_count!"
+  # should we exit fatal, things might not work as expected.
+fi
+
 ### INITIALIZE AND LOG CONFIG VARS ##################################################################################
 log_green "Configuration Options are:
-  TESLA_VIN=$TESLA_VIN1; $TESLA_VIN2; $TESLA_VIN3
-  PRESENCE_DETECTION=$PRESENCE_DETECTION
-  BLE_MAC=$BLE_MAC1; $BLE_MAC2; $BLE_MAC3
+  BLE_MAC_LIST=$BLE_MAC_LIST
+  DEBUG=$DEBUG
   MQTT_IP=$MQTT_IP
   MQTT_PORT=$MQTT_PORT
-  MQTT_USER=$MQTT_USER
   MQTT_PWD=Not Shown
+  MQTT_USER=$MQTT_USER
+  PRESENCE_DETECTION_TTL=$PRESENCE_DETECTION_TTL
   SEND_CMD_RETRY_DELAY=$SEND_CMD_RETRY_DELAY
-  DEBUG=$DEBUG"
+  TESLA_VIN=$TESLA_VIN"
```

Comment:
TESLA_VIN_LIST?

Reply:
Yes indeed , fixed ~5m ago. Thank you for double checking.